src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -4113,6 +4113,16 @@ void Unit::RemoveArenaAuras()
     }
 }
 
+void Unit::RemoveAurasOnEvade()
+{
+    if (IsPet() || IsGuardian()) // if pet or guardian shouldn't lose any aura
```

Comment:
I wonder if this should maybe extend to all player owned creatures, not just hunter/warlock pets and guardians